components/monitoring/base/configure-prometheus.yaml

```diff
@@ -0,0 +1,223 @@
+---
+apiVersion: monitoring.coreos.com/v1
+kind: Prometheus
+metadata:
+  namespace: appstudio-workload-monitoring
+  name: prometheus
+  labels:
+    provider: appstudio-kcp-workload
+    app.kubernetes.io/instance: prometheus
+spec:
+  alerting:
+    alertmanagers:
+    - name: alertmanager-operated
+      namespace: appstudio-workload-monitoring
+      port: web
+      scheme: http
+  image: quay.io/prometheus/prometheus:v2.39.1
+  externalLabels:
+    cluster: appstudio-kcp-workload
+  logLevel: debug
+  replicas: 2
+  retention: 45d
+  walCompression: true
+  ruleSelector:
+    matchLabels:
+      prometheus: appstudio-kcp-workload
+  securityContext: {}
+  serviceAccountName: prometheus-k8s
+  serviceMonitorSelector:
+    matchLabels:
+      prometheus: appstudio-kcp-workload
+  ruleNamespaceSelector:
+    matchLabels:
+      openshift.io/workload-monitoring: "true"
+  storage:
+    volumeClaimTemplate:
+      apiVersion: v1
+      kind: PersistentVolumeClaim
+      spec:
+        accessModes:
+        - ReadWriteOnce
+        resources:
+          requests:
+            storage: 50Gi
+        storageClassName: gp2
+  version: v2.39.1
+  volumeMounts:
+    - mountPath: /etc/prometheus/tls
+      name: secret-prometheus-oauth2-tls
+  volumes:
+  - name: secret-prometheus-oauth2-tls
+    secret:
+      secretName: prometheus-oauth2-tls
+  - name: secret-prometheus-oauth-tls
+    secret:
+      secretName: prometheus-oauth-tls
+  containers:
+  # extra container to have an access to the Prometheus console to human users who need to login via GitHub
+  - name: oauth2-proxy
+    args:
+    - --provider=github
+    - --https-address=:9091
+    - --email-domain=*
+    - --upstream=http://localhost:9090
+    - --tls-cert-file=/etc/tls/private/tls.crt
+    - --tls-key-file=/etc/tls/private/tls.key
+    - --skip-auth-regex=^/metrics
+    - --skip-auth-regex=^/ping
+    - --github-org=redhat-appstudio-sre
+    - --github-team=stage
+    env:
+    - name: OAUTH2_PROXY_CLIENT_ID
+      valueFrom:
+        secretKeyRef:
+          key: client-id
+          name: prometheus-proxy-config
+    - name: OAUTH2_PROXY_CLIENT_SECRET
+      valueFrom:
+        secretKeyRef:
+          key: client-secret
+          name: prometheus-proxy-config
+    - name: OAUTH2_PROXY_COOKIE_SECRET
+      valueFrom:
+        secretKeyRef:
+          key: cookie-secret
+          name: prometheus-proxy-config
+    image: quay.io/pusher/oauth2_proxy:v6.1.1
```

Comment:
you could use quay.io/oauth2-proxy/oauth2-proxy:v7.2.1 (but not 7.3 or 7.4 because the support for GitHub is broken apparently - see https://github.com/oauth2-proxy/oauth2-proxy/issues/1724)

Reply:
Yes I also saw the quay.io/pusher/oauth2_proxy:v6.1.1 image is got getting updated for a long time. Let me update it.